emsdk.py

```diff
@@ -2510,13 +2510,24 @@ def adjusted_path(tools_to_activate, system=False):
     existing_path = os.environ['PATH'].split(ENVPATH_SEPARATOR)
   emsdk_root_path = to_unix_path(emsdk_path())
 
-  existing_emsdk_tools = [i for i in existing_path if to_unix_path(i).startswith(emsdk_root_path)]
-  new_emsdk_tools = [i for i in path_add if not normalized_contains(existing_emsdk_tools, i)]
+  existing_emsdk_tools = []
+  existing_nonemsdk_path = []
+  for entry in existing_path:
+    if to_unix_path(entry).startswith(emsdk_root_path):
+      existing_emsdk_tools.append(entry)
+    else:
+      existing_nonemsdk_path.append(entry)
+
+  new_emsdk_tools = []
+  kept_emsdk_tools = []
+  for entry in path_add:
+    if not normalized_contains(existing_emsdk_tools, entry):
+      new_emsdk_tools.append(entry)
+    else:
+      kept_emsdk_tools.append(entry)
```

Comment:
I'm not quite sure what the purpose of this loop is.    Its seems that it takes `path_add` and splits it into two lists but those lists then both used in their entirety below.. 

How is this different to just doing `whole_path = unique_items(new_path + existing_nonemsdk_path)` below and skipping this loop?   I guess ordering would be the only different?  Do we care about that?

Reply:
new_emsdk_tools is returned as the second output of the function

Reply:
Ah I see, thanks!

Still, can we drop `kept_emsdk_tools` completely and use `whole_path = unique_items(new_path + existing_nonemsdk_path)`?

Reply:
I think it is useful to be explicit. We can use this `kept_emsdk_tools` later to optimize some of the things. The following is true:
```
path_add == new_emsdk_tools + kept_emsdk_tools 
```

The following is what you meant. I have changed the names to prevent any confusion.

```python
whole_path = unique_items(path_add+ existing_nonemsdk_path)
```